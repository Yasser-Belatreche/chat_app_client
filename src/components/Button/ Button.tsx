import React from 'react';
import {Text, TouchableHighlight} from 'react-native';

// styles
import styles from './Button.style';

interface Props {
  onPress?: () => void;
}

const Button: React.FC<Props> = ({onPress, children}) => {
  return (
    <TouchableHighlight style={styles} onPress={onPress}>
      <Text>{children}</Text>
    </TouchableHighlight>
  );
};

export {Button};
