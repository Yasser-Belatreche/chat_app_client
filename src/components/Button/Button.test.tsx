import 'react-native';
import React from 'react';
import rendered from 'react-test-renderer';
import {TouchableHighlight} from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';

import {Button} from './ Button';

describe('Button Component', () => {
  it('should renders correctly', () => {
    const tree = rendered.create(<Button>Click</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should run the function passed in the props when pressed', () => {
    const mockFunction = jest.fn();
    const instance = render(<Button onPress={mockFunction}>Click</Button>);

    fireEvent(instance.UNSAFE_getByType(TouchableHighlight), 'onPress');

    expect(mockFunction).toBeCalledTimes(1);
  });
});
