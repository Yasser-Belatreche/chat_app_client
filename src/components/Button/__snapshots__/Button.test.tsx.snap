// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Text>
    Click
  </Text>
</View>
`;
